### PR TITLE
Rename one letter variables

### PR DESCRIPTION
**Pull request by mkoltonski**
**Description**

So I changed all the variables shown by the command:
`reek --smell UncommunicativeVariableName`
which from my point of view are more understandable😬.
I left only the variables in migrations created by the generators.

#3294
**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)

### CODE REVIEW

**Comment by spaghetticode** [comment]
@mkoltonski thank you for this PR 👏it's much appreciated!

I think the commit messages can be improved, as they are not very communicative about the code changes. 

Also, I'm not sure if all the changes should be squashed in a single commit, or if you prefer to keep them somehow separated (maybe by area? one commit for spree_backend, one for spree_core...). Anyway, as long as messages are expressive and can be helpful in understanding the history of changes then any solution is going to be fine 👍

If you're not sure about what I mean, [this post](https://chris.beams.io/posts/git-commit/) may shed some lights on the matter of communicative and well-written git commits in general.

File: core/app/helpers/spree/base_helper.rb [hunk 1]
```diff
@@ -79,16 +79,16 @@ def taxon_breadcrumbs(taxon, separator = '&nbsp;&raquo;&nbsp;', breadcrumb_class
...
       if taxon
-        crumbs += taxon.ancestors.collect { |a| [a.name, spree.nested_taxons_path(a.permalink)] } unless taxon.ancestors.empty?
+        crumbs += taxon.ancestors.collect { |element| [element.name, spree.nested_taxons_path(element.permalink)] } unless taxon.ancestors.empty?
         crumbs << [taxon.name, spree.nested_taxons_path(taxon.permalink)]
       end
 
       separator = raw(separator)
 
-      items = crumbs.each_with_index.collect do |crumb, i|
+      items = crumbs.each_with_index.collect do |crumb, item|
```
Comment (dportalesr) [inline core/app/helpers/spree/base_helper.rb]: I think `index` better reflects the purpose of the variable than `item`.

File: core/app/models/spree/calculator/shipping/flexi_rate.rb [hunk 2]
```diff
@@ -18,9 +18,9 @@ def compute_package(package)
       def compute_from_quantity(quantity)
         sum = 0
         max = preferred_max_items.to_i
-        quantity.times do |i|
+        quantity.times do |item|
```
Comment (dportalesr) [inline core/app/models/spree/calculator/shipping/flexi_rate.rb]: Same as before: `item` is actually a counter/index here.

File: core/app/models/spree/promotion/actions/create_quantity_adjustments.rb [hunk 3]
```diff
@@ -67,7 +67,7 @@ def compute_amount(line_item)
           line_items = actionable_line_items(order)
 
           actioned_line_items = order.line_item_adjustments.reload.
-            select { |a| a.source == self && a.amount < 0 }.
+            select { |item| item.source == self && item.amount < 0 }.
```
Comment (dportalesr) [inline core/app/models/spree/promotion/actions/create_quantity_adjustments.rb]: I'd use `adjustment` here.

File: core/app/models/spree/zone.rb [hunk 4]
```diff
@@ -40,7 +40,7 @@ class Zone < Spree::Base
     end
 
     alias :members :zone_members
-    accepts_nested_attributes_for :zone_members, allow_destroy: true, reject_if: proc { |a| a['zoneable_id'].blank? }
+    accepts_nested_attributes_for :zone_members, allow_destroy: true, reject_if: proc { |attribute| attribute['zoneable_id'].blank? }
```
Comment (dportalesr) [inline core/app/models/spree/zone.rb]: Probably `zone_member` describes better this variable.

File: core/lib/spree/core/importer/order.rb [hunk 5]
```diff
@@ -170,8 +170,8 @@ def self.ensure_country_id_from_params(address)
             search[:iso_name] = iso_name.upcase
           elsif iso = address[:country]['iso']
             search[:iso] = iso.upcase
-          elsif iso3 = address[:country]['iso3']
-            search[:iso3] = iso3.upcase
+          elsif iso_three = address[:country]['iso_three']
```
Comment (dportalesr) [inline core/lib/spree/core/importer/order.rb]: I suspect this key name change was a mistake.

File: core/lib/spree/core/importer/order.rb [hunk 6]
```diff
@@ -170,8 +170,8 @@ def self.ensure_country_id_from_params(address)
...
+            search[:iso_three] = iso_three.upcase
```
Comment (dportalesr) [inline core/lib/spree/core/importer/order.rb]: Probably here too: `search[:iso_3]`

File: core/spec/models/spree/stock/estimator_spec.rb [hunk 7]
```diff
@@ -8,8 +8,8 @@ module Stock
       let(:shipping_rate) { 4.00 }
       let!(:shipping_method) { create(:shipping_method, cost: shipping_rate, currency: currency) }
       let(:package) do
-        build(:stock_package, contents: inventory_units.map { |i| ContentItem.new(i) }).tap do |p|
-          p.shipment = p.to_shipment
+        build(:stock_package, contents: inventory_units.map { |item| ContentItem.new(item) }).tap do |place|
```
Comment (dportalesr) [inline core/spec/models/spree/stock/estimator_spec.rb]: I think `place` is misleading here. I suggest using `package`.

File: frontend/lib/generators/solidus/views/override_generator.rb [hunk 8]
```diff
@@ -33,8 +33,8 @@ def copy_views
 
       def views_to_copy
         if @options['only']
-          VIEWS.select do |v|
-            Pathname.new(v).relative_path_from(source_dir).to_s.include?(@options['only'])
+          VIEWS.select do |value|
```
Comment (dportalesr) [inline frontend/lib/generators/solidus/views/override_generator.rb]: I suggest using `view` here.

File: guides/helpers/image_helpers.rb [hunk 9]
```diff
@@ -18,7 +18,7 @@ def inline_svg(filename, options = {})
     asset = "source/assets/images/#{filename}"
 
     if File.exist?(asset)
-      file = File.open(asset, 'r') { |f| f.read }
+      file = File.open(asset, 'r') { |value| value.read }
```
Comment (dportalesr) [inline guides/helpers/image_helpers.rb]: Why not use `file` instead of `value`?

**Comment by tvdeyen** [comment]
Superseded by #3400

**Comment by tvdeyen** [comment]
Thanks for all the hard work, though. It is incorperated in #3400 :+1:

**Pull request closed without merging**